sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/TemporalMetricStorage.java

```diff
@@ -74,7 +74,20 @@ synchronized MetricData buildMetricFor(
       } else if (temporality == AggregationTemporality.CUMULATIVE && isSynchronous) {
         // We need to make sure the current delta recording gets merged into the previous cumulative
         // for the next cumulative measurement.
-        MetricStorageUtils.mergeInPlace(last.getAccumulation(), currentAccumulation, aggregator);
+
+        // We select whether we "clean" un-seen measurements based on how many we had in the last
+        // cycle.
+        if (last.getAccumulation().size() >= MetricStorageUtils.MAX_ACCUMULATIONS) {
+          MetricStorageUtils.mergeInPlace(last.getAccumulation(), currentAccumulation, aggregator);
```

Comment:
Is this necessary? Doesn't lines 87-89 take care of this since `currentAccumulation.size()` will always be <= `MetricStorageUtils.MAX_ACCUMULATIONS`?

Reply:
Good point, added this after writing the test, fixed!